src/scrollspy.ts

```diff
@@ -250,3 +301,30 @@ export class ScrollSpy extends Component<ScrollSpyOptions> {
     ScrollSpy._ticks = 0;
   }
 }
+
+function getDistanceToViewport(element) {
+  const rect = element.getBoundingClientRect();
+  const distance = rect.top;
+  return distance;
+}
+
+function smoothScrollIntoView(element, duration = 300) {
+  const targetPosition = element.getBoundingClientRect().top + (window.scrollY || window.pageYOffset);
+  const startPosition = (window.scrollY || window.pageYOffset);
+  const distance = targetPosition - startPosition;
+  const startTime = performance.now();
+
+  function scrollStep(currentTime) {
+    const elapsed = currentTime - startTime;
+    const progress = Math.min(elapsed / duration, 1);
+    const scrollY = startPosition + distance * progress;
+
+    if (progress < 1) {
+      window.scrollTo(0, scrollY);
+      requestAnimationFrame(scrollStep);
+    } else {
+      window.scrollTo(0, targetPosition);
+    }
+  }
+  requestAnimationFrame(scrollStep);
+}
```

Comment:
i think this two functions should be inside of the class. The rest looks nice!
Can you place them as private functions, then i will merge. Thanks

Reply:
:heavy_check_mark: